### ISSUE TITLE
Checksum mismatch

### ISSUE THREAD

**Issue by christianhuening** [hashicorp/go-rootcerts#4]
We compiled a golang programm using this repository. As we're using go modules the checksum for this library was already in the go.sum file. On a new setup a new `go build` run resulted in a checksum mismatch:

```
go: verifying github.com/hashicorp/go-rootcerts@v0.0.0-20160503143440-6bb64b370b90: checksum mismatch
	downloaded: h1:VBj0QYQ0u2MCJzBfeYXGexnAl17GsH1yidnoxCqqD9E=
	go.sum:     h1:9HVkPxOpo+yO93Ah4yrO67d/qh0fbLLWbKqhYjyHq9A=
```

Question: How can that happen?

**Comment by arifsetiawan** [comment]
I have this issue too. how we can fix this?

**Comment by al4** [comment]
Found this via web search while researching a similar problem. In my case there was a mismatch in Go versions; they changed how symlinks are handled between 1.11.3 and 1.11.4, which alters the checksums. Thus a go.sum file generated with go 1.11.3 is likely to fail on go 1.11.4 and vice versa. As I was using `golang:1.11-alpine3.8` in the Dockerfile, and go 1.11.4 on my laptop, the solution was to `docker pull golang:1.11-alpine3.8` to update the image. Or fix to `golang:1.11.4-alpine3.8` in the Dockerfile.

Context: https://github.com/golang/go/issues/29278

**Comment by jefferai** [comment]
Thanks @al4 for the info! As far as I can tell this shouldn't be an issue anymore, so I'm closing it. If it's still a problem (and you're on go 1.11.4 and per the bug linked above you've done a `go clean -modcache` then please report back.

**Comment by arifsetiawan** [comment]
For me this happens when running build on gitlab CI. I use golang 1.12. I think the mod cache is as clean as it can get. The fix is to go get with specific commit hash instead of using default.

```
go get github.com/hashicorp/go-rootcerts@6bb64b370b90e7ef1fa532be9e591a81c3493e00
```

**Comment by thomloftus** [comment]
I'm having this exact same issue in Go 1.12.5 darwin/amd64.

go clean -modcache doesn't fix it.

locally, I get:
```
verifying github.com/hashicorp/go-rootcerts@v0.0.0-20160503143440-6bb64b370b90: checksum mismatch
	downloaded: h1:VBj0QYQ0u2MCJzBfeYXGexnAl17GsH1yidnoxCqqD9E=
	go.sum:     h1:9HVkPxOpo+yO93Ah4yrO67d/qh0fbLLWbKqhYjyHq9A=
```
so if I then remove the offending line from my go.sum and re-build, all is well.

If I then PR the change to my go.sum file, it fails in Gitlab with the exact opposite checksum mismatch:

```
verifying github.com/hashicorp/go-rootcerts@v0.0.0-20160503143440-6bb64b370b90: checksum mismatch
	downloaded: h1:9HVkPxOpo+yO93Ah4yrO67d/qh0fbLLWbKqhYjyHq9A=
	go.sum:     h1:VBj0QYQ0u2MCJzBfeYXGexnAl17GsH1yidnoxCqqD9E=
```

it would appear that if my go.sum specifies the checksum beginning with VBj, that it should in fact download the same based on:
https://github.com/hashicorp/packer/blob/master/go.sum#L184

very perplexed...

**Comment by tamsky** [comment]
Also seeing this in `golang:1.12.5-alpine3.9` on `linux/amd64`

**Comment by subodhp** [comment]
Also present in go version go1.12.6 linux/amd64

**Comment by jefferai** [comment]
With which tag?

**Comment by subodhp** [comment]
verifying github.com/hashicorp/go-rootcerts@v1.0.0: checksum mismatch
        downloaded: h1:Rqb66Oo1X/eSV1x66xbDccZjhJigjg0+e82kpwzSwCI=
        go.sum:     h1:ueI78wUjYExhCvMLow4icJnayNNFRgy0d9EGs/a1T44=

**Comment by jefferai** [comment]
Please try 1.0.1

**Comment by subodhp** [comment]
Thanks @jefferai , it is fixed with 1.0.1.
